internal/cluster/distribution/eks/eksprovider/workflow/update_eks_version_activity.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+const UpdateClusterVersionActivityName = "eks-update-version"
+
+// UpdateClusterVersionActivity responsible for updating an EKS cluster
+type UpdateClusterVersionActivity struct {
+	awsSessionFactory AWSFactory
+}
+
+// UpdateClusterVersionActivityInput holds data needed for updating an EKS cluster
+type UpdateClusterVersionActivityInput struct {
+	EKSActivityInput
+
+	Version string
+}
+
+// UpdateClusterVersionActivityOutput holds the output data of the UpdateEKSVersionActivityOutput
+type UpdateClusterVersionActivityOutput struct {
+	UpdateID string
+}
+
+// NewUpdateClusterVersionActivity instantiates a new EKS version update
+func NewUpdateClusterVersionActivity(awsSessionFactory AWSFactory) *UpdateClusterVersionActivity {
+	return &UpdateClusterVersionActivity{
+		awsSessionFactory: awsSessionFactory,
+	}
+}
+
+func (a *UpdateClusterVersionActivity) Execute(ctx context.Context, input UpdateClusterVersionActivityInput) (*UpdateClusterVersionActivityOutput, error) {
+	session, err := a.awsSessionFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return nil, err
+	}
+
+	eksClient := eks.New(session)
```

Comment:
**Optional**: we **MAY** consider using an EKS API interface and a corresponding factory here for better mocking capability and thus testing capability.
The `eksinterface.EKSAPI` can be mocked locally similarly to how the `cloudformation.go.cloudFormationAPI` is done and a factory analogous to the `cloudformation.go.CloudFormationAPIFactory` can be added to the activity.

Same at:
https://github.com/banzaicloud/pipeline/pull/3123/files#diff-ad1a4a93014f9f6fe3506f142f5a11ffR53

Reply:
Good point, I will add this.

Reply:
Thanks for the fix.